x-pack/plugin/vector-tile/src/test/java/org/elasticsearch/xpack/vectortile/feature/FeatureFactoryTests.java

```diff
@@ -230,7 +244,7 @@ public void testManyIntersectingGeometries() {
                 GeoTileUtils.toBoundingBox(2 * x + 1, 2 * y + 1, z + 1)
             )
         );
-        assertThat(builder.getFeatures(withinCollection), iterableWithSize(4));
+        assertThat(builder.getFeatures(withinCollection), iterableWithSize(1));
```

Comment:
When debugging I see that a GEOMETRYCOLLECTION was converted into a MULTIPOLYGON. Is this desired behaviour?

Reply:
Because we are more clever now and we create a multi-polygon instead of a geometry collection. I think it is an improvement as it makes the tile smaller.

Reply:
Maybe we should add a comment and check that the type we get is a POLYGON?

Reply:
>Is this desired behaviour?
I think it is a good change because makes the vector tile smaller

Reply:
I took a quick look, and since the type returned is MVT binary, we would need to do a little extra decoding to make the assertions. It would be nice though.